Core/include/Acts/Propagator/MultiEigenStepperLoop.ipp

```diff
@@ -37,8 +37,10 @@ auto MultiEigenStepperLoop<E, R, A>::boundState(
                                           transportCov, freeToBoundCorrection);
 
       if (bs.ok()) {
-        states.push_back(
-            {state.components[i].weight, std::get<BoundTrackParameters>(*bs)});
+        const auto& btp = std::get<BoundTrackParameters>(*bs);
+        states.emplace_back(
+            state.components[i].weight, btp.parameters(),
+            btp.covariance().value_or(Acts::BoundSymMatrix::Zero()));
```

Comment:
will a zero covariance play nice later on?

Reply:
Hmm I thought if everything is consistent, `covTransport = true` should mean there is also a covariance there to get... However, I think this is not checked...
Maybe worth thinking about how to ensure this does not go to the outside.